### PR TITLE
test: add gen2 e2e tests

### PR DESCRIPTION
**Pull request by AaronZyLee**
#### Description of changes

Added Gen2 e2e tests for `graphql-generator` related command

- `npx ampx generate outputs`
- `npx ampx generate forms`
- `npx ampx generate graphql-client-code`

#### Description of how you validated changes
Unit and E2E tests passed. [Link to E2E](https://tiny.amazon.com/1dc95a1pl/IsenLink)

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-codegen/blob/main/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] Breaking changes to existing customers are released behind a feature flag or major version update
- [ ] Changes are tested using sample applications for all relevant platforms (iOS/android/flutter/Javascript) that use the feature added/modified
- [ ] Changes are tested on windows. Some Node functions (such as `path`) behave differently on windows.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-codegen-e2e-tests/src/__tests__/graphql-generator-gen2.test.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    const statementTargets = ['javascript', 'graphql', 'flow', 'typescript', 'angular'];
+    const typeTargets = ['json', 'swift', 'typescript', 'flow', 'scala', 'flow-modern', 'angular'];
+    const typeTargetConfigs = typeTargets.map(tt => { return { outDir: 'codegen', format: 'graphql-codegen', typeTarget: tt }});
+    const graphqlCodegenConfigs: GraphqlCodegenConfig[] = statementTargets.map(st => {
+      return typeTargetConfigs.map(config => {
+        return { ...config, statementTarget: st } as GraphqlCodegenConfig
+      });
+    }).flat();
+    graphqlCodegenConfigs.forEach(config => {
+      // TODO: skip these tests as it will fail due to the duplicate graphql module. Will enable them once the issue is resolved
```
Comment (AaronZyLee) [inline packages/amplify-codegen-e2e-tests/src/__tests__/graphql-generator-gen2.test.ts:59]: These tests fail due to the duplicate graphql module detected. Will explore if there are any workarounds before going to the verdaccio registry. This will be addressed in a separate PR.

**Review by phani-srikar — APPROVED** [review]

File: packages/amplify-codegen-e2e-tests/src/__tests__/graphql-generator-gen2.test.ts (hunk 1 above)
Comment (phani-srikar) [reply packages/amplify-codegen-e2e-tests/src/__tests__/graphql-generator-gen2.test.ts:59]: There is a [follow-up task for this](https://app.asana.com/0/1206048977824382/1207233076780798/f)

**Review by palpatim — APPROVED** [review]
